From tools/ unittests, some style violations to be cleaned up:

```
./wptrunner/wptrunner/browsers/android_webview.py:1:1: F401 'subprocess' imported but unused
./wptrunner/wptrunner/browsers/android_webview.py:3:1: F401 '.base.ExecutorBrowser' imported but unused
./wptrunner/wptrunner/browsers/android_webview.py:6:1: F401 '.chrome_android.AndroidBrowser' imported but unused
./wptrunner/wptrunner/browsers/android_webview.py:7:1: F401 '..webdriver_server.ChromeDriverServer' imported but unused
./wptrunner/wptrunner/browsers/android_webview.py:99:73: W292 no newline at end of file
./wptrunner/wptrunner/browsers/android_weblayer.py:1:1: F401 'subprocess' imported but unused
./wptrunner/wptrunner/browsers/android_weblayer.py:3:1: F401 '.base.Browser' imported but unused
./wptrunner/wptrunner/browsers/android_weblayer.py:3:1: F401 '.base.ExecutorBrowser' imported but unused
./wptrunner/wptrunner/browsers/android_weblayer.py:7:1: F401 '..webdriver_server.ChromeDriverServer' imported but unused
./wptrunner/wptrunner/browsers/android_weblayer.py:101:73: W292 no newline at end of file
./wptrunner/wptrunner/browsers/chrome_android.py:116:9: E722 do not use bare 'except'
```

--- a/tools/wptrunner/wptrunner/browsers/android_weblayer.py
+++ b/tools/wptrunner/wptrunner/browsers/android_weblayer.py
@@ -1,9 +1,7 @@
-import subprocess
-
-from .base import Browser, ExecutorBrowser, require_arg
+from .base import require_arg
 from .base import get_timeout_multiplier   # noqa: F401
 from .chrome import executor_kwargs as chrome_executor_kwargs
-from .chrome_android import AndroidBrowser
+from .chrome_android import ChromeAndroidBrowserBase
 from ..webdriver_server import ChromeDriverServer
 from ..executors.executorwebdriver import (WebDriverTestharnessExecutor,  # noqa: F401
                                            WebDriverRefTestExecutor)  # noqa: F401
@@ -75,7 +73,7 @@ def env_options():
     return {"server_host": "127.0.0.1"}
 
 
-class WeblayerShell(AndroidBrowser):
+class WeblayerShell(ChromeAndroidBrowserBase):
     """Chrome is backed by chromedriver, which is supplied through
     ``wptrunner.webdriver.ChromeDriverServer``.
     """
@@ -90,12 +88,5 @@ class WeblayerShell(AndroidBrowser):
         super(WeblayerShell, self).__init__(logger,
                 webdriver_binary, remote_queue, device_serial, webdriver_args)
         self.binary = binary
+        self.wptserver_ports = _wptserve_ports
 
-    def setup_adb_reverse(self):
-        self._adb_run(['wait-for-device'])
-        self._adb_run(['forward', '--remove-all'])
-        self._adb_run(['reverse', '--remove-all'])
-        # "adb reverse" basically forwards network connection from device to
-        # host.
-        for port in _wptserve_ports:
-            self._adb_run(['reverse', 'tcp:%d' % port, 'tcp:%d' % port])

--- a/tools/wptrunner/wptrunner/browsers/android_weblayer.py
+++ b/tools/wptrunner/wptrunner/browsers/android_weblayer.py
@@ -2,7 +2,6 @@ from .base import require_arg
 from .base import get_timeout_multiplier   # noqa: F401
 from .chrome import executor_kwargs as chrome_executor_kwargs
 from .chrome_android import ChromeAndroidBrowserBase
-from ..webdriver_server import ChromeDriverServer
 from ..executors.executorwebdriver import (WebDriverTestharnessExecutor,  # noqa: F401
                                            WebDriverRefTestExecutor)  # noqa: F401
 from ..executors.executorchrome import ChromeDriverWdspecExecutor  # noqa: F401

--- a/tools/wptrunner/wptrunner/browsers/android_webview.py
+++ b/tools/wptrunner/wptrunner/browsers/android_webview.py
@@ -2,7 +2,6 @@ from .base import require_arg
 from .base import get_timeout_multiplier   # noqa: F401
 from .chrome import executor_kwargs as chrome_executor_kwargs
 from .chrome_android import ChromeAndroidBrowserBase
-from ..webdriver_server import ChromeDriverServer
 from ..executors.executorwebdriver import (WebDriverTestharnessExecutor,  # noqa: F401
                                            WebDriverRefTestExecutor)  # noqa: F401
 from ..executors.executorchrome import ChromeDriverWdspecExecutor  # noqa: F401

--- a/tools/wptrunner/wptrunner/browsers/android_webview.py
+++ b/tools/wptrunner/wptrunner/browsers/android_webview.py
@@ -1,9 +1,7 @@
-import subprocess
-
-from .base import Browser, ExecutorBrowser, require_arg
+from .base import require_arg
 from .base import get_timeout_multiplier   # noqa: F401
 from .chrome import executor_kwargs as chrome_executor_kwargs
-from .chrome_android import AndroidBrowser
+from .chrome_android import ChromeAndroidBrowserBase
 from ..webdriver_server import ChromeDriverServer
 from ..executors.executorwebdriver import (WebDriverTestharnessExecutor,  # noqa: F401
                                            WebDriverRefTestExecutor)  # noqa: F401
@@ -74,7 +72,7 @@ def env_options():
     return {"server_host": "127.0.0.1"}
 
 
-class SystemWebViewShell(Browser):
+class SystemWebViewShell(ChromeAndroidBrowserBase):
     """Chrome is backed by chromedriver, which is supplied through
     ``wptrunner.webdriver.ChromeDriverServer``.
     """
@@ -88,12 +86,5 @@ class SystemWebViewShell(Browser):
         super(SystemWebViewShell, self).__init__(logger,
                 webdriver_binary, remote_queue, device_serial, webdriver_args)
         self.binary = binary
+        self.wptserver_ports = _wptserve_ports
 
-    def setup_adb_reverse(self):
-        self._adb_run(['wait-for-device'])
-        self._adb_run(['forward', '--remove-all'])
-        self._adb_run(['reverse', '--remove-all'])
-        # "adb reverse" basically forwards network connection from device to
-        # host.
-        for port in _wptserve_ports:
-            self._adb_run(['reverse', 'tcp:%d' % port, 'tcp:%d' % port])

--- a/tools/wptrunner/wptrunner/browsers/chrome_android.py
+++ b/tools/wptrunner/wptrunner/browsers/chrome_android.py
@@ -83,18 +83,24 @@ class LogcatRunner(object):
 
     def _run(self):
         try:
+            # TODO: adb logcat -c fail randomly with message
+            # "failed to clear the 'main' log"
             self.browser.clear_log()
-            self._cmd = self.browser.logcat_cmd()
-            self._proc = mozprocess.ProcessHandler(
-                self._cmd,
-                processOutputLine=self.on_output,
-                storeOutput=False)
-            self._proc.run()
-        except (OSError, subprocess.CalledProcessError):
-            self.logger.error("Failed to start adb logcat")
+        except subprocess.CalledProcessError:
+            self.logger.error("Failed to clear logcat buffer")
+
+        self._cmd = self.browser.logcat_cmd()
+        self._proc = mozprocess.ProcessHandler(
+            self._cmd,
+            processOutputLine=self.on_output,
+            storeOutput=False)
+        self._proc.run()
 
     def _send_message(self, command, *args):
-        self.remote_queue.put((command, args))
+        try:
+            self.remote_queue.put((command, args))
+        except AssertionError:
+            self.logger.warning("Error when send to remote queue")
 
     def stop(self, force=False):
         if self.is_alive():
@@ -106,33 +112,30 @@ class LogcatRunner(object):
         return hasattr(self._proc, "proc") and self._proc.poll() is None
 
     def on_output(self, line):
-        try:
-            data = {
-                "process": "LOGCAT",
-                "command": "logcat",
-                "data": line
-            }
-            self._send_message("log", "process_output", data)
-        except:
-            pass
+        data = {
+            "process": "LOGCAT",
+            "command": "logcat",
+            "data": line
+        }
+        self._send_message("log", "process_output", data)
 
-class AndroidBrowser(Browser):
+class ChromeAndroidBrowserBase(Browser):
     def __init__(self, logger,
                  webdriver_binary="chromedriver",
                  remote_queue = None,
                  device_serial=None, webdriver_args=None):
-        super(AndroidBrowser, self).__init__(logger)
+        super(ChromeAndroidBrowserBase, self).__init__(logger)
         self.device_serial = device_serial
         self.remote_queue = remote_queue
         self.server = ChromeDriverServer(self.logger,
                                          binary=webdriver_binary,
                                          args=webdriver_args)
-        self.setup_adb_reverse()
         if self.remote_queue is not None:
             self.logcat_runner = LogcatRunner(self.logger,
                                           self, self.remote_queue)
 
     def setup(self):
+        self.setup_adb_reverse()
         if self.remote_queue is not None:
             self.logcat_runner.start()
 
@@ -143,9 +146,6 @@ class AndroidBrowser(Browser):
         cmd.extend(args)
         self.logger.info(' '.join(cmd))
         subprocess.check_call(cmd)
-
-    def setup_adb_reverse(self):
-        pass
 
     def start(self, **kwargs):
         self.server.start(block=False)
@@ -182,7 +182,15 @@ class AndroidBrowser(Browser):
         cmd.extend(['logcat', '*:D'])
         return cmd
 
-class ChromeAndroidBrowser(AndroidBrowser):
+    def setup_adb_reverse(self):
+        self._adb_run(['wait-for-device'])
+        self._adb_run(['forward', '--remove-all'])
+        self._adb_run(['reverse', '--remove-all'])
+        # "adb reverse" forwards network connection from device to host.
+        for port in self.wptserver_ports:
+            self._adb_run(['reverse', 'tcp:%d' % port, 'tcp:%d' % port])
+
+class ChromeAndroidBrowser(ChromeAndroidBrowserBase):
     """Chrome is backed by chromedriver, which is supplied through
     ``wptrunner.webdriver.ChromeDriverServer``.
     """
@@ -194,11 +202,5 @@ class ChromeAndroidBrowser(AndroidBrowser):
         super(ChromeAndroidBrowser, self).__init__(logger,
                 webdriver_binary, remote_queue, device_serial, webdriver_args)
         self.package_name = package_name
+        self.wptserver_ports = _wptserve_ports
 
-    def setup_adb_reverse(self):
-        self._adb_run(['wait-for-device'])
-        self._adb_run(['forward', '--remove-all'])
-        self._adb_run(['reverse', '--remove-all'])
-        # "adb reverse" forwards network connection from device to host.
-        for port in _wptserve_ports:
-            self._adb_run(['reverse', 'tcp:%d' % port, 'tcp:%d' % port])

--- a/tools/wptrunner/wptrunner/formatters/chromium.py
+++ b/tools/wptrunner/wptrunner/formatters/chromium.py
@@ -272,6 +272,6 @@ class ChromiumFormatter(base.BaseFormatter):
         return json.dumps(final_result)
 
     def process_output(self, data):
-        cmd = data.get('command', "")
+        cmd = data.get("command", "")
         if any(c in cmd for c in ["chromedriver", "logcat"]):
             self.browser_log.append(data['data'])

